WalletWasabi.Tests/UnitTests/BlockchainAnalysis/CoinJoinAnonscoreTests.cs

```diff
@@ -167,9 +167,12 @@ public void InputSanityBeforeSelfAnonsetSanityCheck()
 
 			Assert.Equal(1, tx.WalletInputs.First().HdPubKey.AnonymitySet);
 
-			// The anonset calculation should be 5, but there's 4 inputs so - 1 = 3.
-			// After than - 1 because 2 out of it is ours.
-			Assert.All(tx.WalletOutputs, x => Assert.Equal(2, x.HdPubKey.AnonymitySet));
+			// The anonset calculation naively would be 5,
+			// but there's only 3 inputs so that limits our anonset to 3.
+			// After that we should get 3/2 because 2 out of 3 is ours.
+			// Finally we don't mess around with decimal precisions, so
+			// conservatively 3/2 = 1.
+			Assert.All(tx.WalletOutputs, x => Assert.Equal(1, x.HdPubKey.AnonymitySet));
```

Comment:
I'm not sure this is correct. I'm afraid this is not imprecision, but rather inaccuracy. If lots of users join the anonymity set with multiple of their coins, then we unnecessarily punish the anonymity score of the CoinJoin by rounding down the corresponding anonymity sets.

This phenomenon also shows the limitations of the k-anonymity privacy model. A rational number as the size of the anonymity set just sounds weird. Not to mention, that it looks horrible in the UI. Probabilities (and therefore entropy-like anon-scores) are more natural in this context as well. However, obviously, such metrics should not be shown on the UI, but they are inevitable for accurately capturing privacy.

Reply:
This is imprecision for sure. 3/2 = 1 for integers. And the calculation is correct, see the other test.

> If lots of users join the anonymity set with multiple of their coins, then we unnecessarily punish the anonymity score of the CoinJoin by rounding down the corresponding anonymity sets.

There is no anonymity score of the coinjoin. This is the main Wasabi repo, no the WabiSabi simulation :)

Here only users have anonymity score and the maximum imprecision is like 1 whole number, which if we start caring that means we're working on something insignificant and we're wasting our time with that.